### PR TITLE
Security Event and Windows Event User management

### PR DESCRIPTION
**Pull request by Alekhya0824**
Required items, please complete
   
   Change(s):
   - Created a new file for Windows Event and added Windows parser in it. In Security Event file removed Windows Event parser and in ASimUserManagement added Windows Event Parser name  and same changes have done for vim as well.

   Reason for Change(s):
   - To Split the Security Event and Windows Event parser separately

   Version Updated:
   - yes

   Testing Completed:
   - yes

   Checked that the validations are passing and have addressed any issues that are present:
   - yes

### CODE REVIEW

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: User Management ASIM parser for Microsoft Security Event logs
-  Version: '0.1.0'
-  LastUpdated: 16 Jul, 2023
+  Version: '0.2.1'
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml:3]: 0.2.0

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: User Management ASIM parser for Microsoft Security Event logs
-  Version: '0.1.1'
-  LastUpdated: Mar 07, 2024
+  Version: '0.2.1'
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml:3]: 0.2.0

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml [hunk 3]
```diff
@@ -0,0 +1,160 @@
+Parser:
+  Title: User Management ASIM parser for Microsoft Security Event logs
+  Version: '0.2.1'
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:3]: 0.2.0

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml [hunk 4]
```diff
@@ -0,0 +1,160 @@
+Parser:
+  Title: User Management ASIM parser for Microsoft Security Event logs
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:2]: Windows Event logs?

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
+  LastUpdated: 28 May, 2024
 Product:
   Name: Microsoft
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml:6]: Microsoft Security Event?

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml (hunk 4 above)
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:2]: Windows Event logs?

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+  Version: '0.2.1'
+  LastUpdated: May 29, 2024
+Product:
+  Name: Microsoft
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:6]: Microsoft Windows Event?

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+  Version: '0.1.1'
+References:
+- Title: ASIM User Management Schema
+  Link: https://aka.ms/ASimUserManagementDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: Audit User Account Management
+  Link: https://learn.microsoft.com/en-us/windows/security/threat-protection/auditing/audit-user-account-management
+Description: |
+  This ASIM parser supports normalizing Microsoft Security Event logs ingested in 'WindowsEvent' table delivered using AMA to the ASIM UserManagement normalized schema.
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:18]: we sure these logs gets ingested using AMA?

File: Parsers/ASimUserManagement/Parsers/imUserManagement.yaml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: User Management ASIM filtering parser
   Version: '0.1.2'
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/imUserManagement.yaml:3]: please increase this version number to 0.1.3

File: Parsers/ASimUserManagement/Parsers/ASimUserManagement.yaml [hunk 9]
```diff
@@ -39,6 +39,7 @@ ParserQuery: |
       union isfuzzy=true
           vimUserManagementEmpty,
           ASimUserManagementMicrosoftSecurityEvent (ASimBuiltInDisabled or ('ExcludeASimUserManagementMicrosoftSecurityEvent' in (DisabledParsers))),
+          ASimUserManagementMicrosoftWindowsEvent (ASimBuiltInDisabled or ('ExcludeASimUserManagementMicrosoftWindowsEvent' in (DisabledParsers))),
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/ASimUserManagement.yaml:42]: please increase union parser version to 0.1.2

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml:1]: please apply same comments as ASIM parser

File: Parsers/ASimUserManagement/Tests/Security_Event_ASimUserManagement_SchemaTest.csv [hunk 11]
```diff
@@ -0,0 +1,72 @@
+﻿Result
+"(0) Error: Missing field [ActorUserIdType] is mandatory when field [ActorUserId] exists"
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Tests/Security_Event_ASimUserManagement_SchemaTest.csv:2]: can you please check this error and map ActorUserIDType?

File: Parsers/ASimUserManagement/Tests/Security_Event_VimUserManagement_SchemaTest.csv [hunk 12]
```diff
@@ -0,0 +1,72 @@
+﻿Result
+"(0) Error: Missing field [ActorUserIdType] is mandatory when field [ActorUserId] exists"
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Tests/Security_Event_VimUserManagement_SchemaTest.csv:2]: please apply same changes as ASIM parser

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml (hunk 10 above)
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml:1]: 1. Are Column count same with old and new parser?
2. Are row count coming same with old and new parser?
3. Did we do filter testing for new parser for ASIM and VIM?

**Review by vakohl — CHANGES_REQUESTED** [review]

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please check above comments and act accordingly. Thanks

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please work on the open comments of the reviewer. Thanks

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
We are waiting for your reply. Thanks

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Could you please address the reviewer’s open comments?

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
There are a few open comments from the reviewer; please act on them. Thanks

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml [hunk 13]
```diff
@@ -0,0 +1,160 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:1]: can you please test filter for this parser from ASC portal?

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftWindowsEvent.yaml [hunk 14]
```diff
@@ -0,0 +1,242 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftWindowsEvent.yaml:1]: can you please test filter for this parser from ASC portal?

**Review by vakohl — CHANGES_REQUESTED** [review]

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please act on reviewer's open comments. Thanks

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml (hunk 10 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml:1]: yes

File: Parsers/ASimUserManagement/Parsers/ASimUserManagement.yaml (hunk 9 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagement.yaml:43]: done

File: Parsers/ASimUserManagement/Parsers/imUserManagement.yaml (hunk 8 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/imUserManagement.yaml:3]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml (hunk 7 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:18]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml (hunk 6 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:6]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml (hunk 4 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:2]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml (hunk 5 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml:6]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml (hunk 4 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:2]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml (hunk 3 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:3]: done

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml (hunk 2 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml:3]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml (hunk 1 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml:3]: done

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftWindowsEvent.yaml (hunk 14 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftWindowsEvent.yaml:1]: yes

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml (hunk 13 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:1]: done

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
I can see a few open comments from the reviewer. Please act on them.
image
And also, one suggestion: once you're done with the open comment change, please click on the `Resolve Conversation` button

File: Parsers/ASimUserManagement/Tests/Security_Event_ASimUserManagement_SchemaTest.csv (hunk 11 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Tests/Security_Event_ASimUserManagement_SchemaTest.csv:2]: done

File: Parsers/ASimUserManagement/Tests/Security_Event_VimUserManagement_SchemaTest.csv (hunk 12 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Tests/Security_Event_VimUserManagement_SchemaTest.csv:2]: done

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml (hunk 10 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml:1]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml [hunk 15]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: User Management ASIM parser for Microsoft Security Event logs
-  Version: '0.1.0'
-  LastUpdated: 16 Jul, 2023
+  Version: '0.2.1'
+  LastUpdated: 28 May, 2024
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml [hunk 16]
```diff
@@ -15,8 +15,8 @@ References:
 - Title: Audit User Account Management
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml:15]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml [hunk 17]
```diff
@@ -15,8 +15,8 @@ References:
...
   Link: https://learn.microsoft.com/en-us/windows/security/threat-protection/auditing/audit-user-account-management
 Description: |
-  This ASIM parser supports normalizing Microsoft Security Event logs delivered using AMA to the ASIM UserManagement normalized schema.
-ParserName: ASimUserManagementMicrosoftSecurityEvent
+  This ASIM parser supports normalizing Microsoft Security Event logs ingested in 'SecurityEvent' table delivered using AMA to the ASIM UserManagement normalized schema.
+  ParserName: ASimUserManagementMicrosoftSecurityEvent
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftSecurityEvent.yaml]: done

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml [hunk 18]
```diff
@@ -15,7 +15,7 @@ References:
 - Title: Audit User Account Management
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftSecurityEvent.yaml:15]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagement.yaml [hunk 19]
```diff
@@ -39,6 +39,7 @@ ParserQuery: |
       union isfuzzy=true
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagement.yaml:40]: done

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftWindowsEvent.yaml [hunk 20]
```diff
@@ -0,0 +1,242 @@
...
+Product:
+  Name: Microsoft
+Normalization:
+  Schema: UserManagement
+  Version: '0.1.1'
+References:
+- Title: ASIM User Management Schema
+  Link: https://aka.ms/ASimUserManagementDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftWindowsEvent.yaml]: done

File: Parsers/ASimUserManagement/Tests/Windows_Event_ASimUserManagement_SchemaTest.csv [hunk 21]
```diff
@@ -0,0 +1,73 @@
+﻿Result
+"(0) Error: Missing field [ActorUserIdType] is mandatory when field [ActorUserId] exists"
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Tests/Windows_Event_ASimUserManagement_SchemaTest.csv:2]: done

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml [hunk 22]
```diff
@@ -0,0 +1,160 @@
...
+Product:
+  Name: Microsoft
+Normalization:
+  Schema: UserManagement
+  Version: '0.1.1'
+References:
+- Title: ASIM User Management Schema
+  Link: https://aka.ms/ASimUserManagementDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml]: done

File: Parsers/ASimUserManagement/Parsers/imUserManagement.yaml [hunk 23]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Parsers/imUserManagement.yaml:1]: done

File: Parsers/ASimUserManagement/Tests/Windows_Event_VimUserManagement_SchemaTest.csv [hunk 24]
```diff
@@ -0,0 +1,74 @@
+﻿Result
+"(0) Error: Missing field [ActorUserIdType] is mandatory when field [ActorUserId] exists"
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Tests/Windows_Event_VimUserManagement_SchemaTest.csv:2]: done

File: Parsers/ASimUserManagement/Tests/Security_Event_ASimUserManagement_SchemaTest.csv (hunk 11 above)
Comment (vakohl) [reply Parsers/ASimUserManagement/Tests/Security_Event_ASimUserManagement_SchemaTest.csv]: @Alekhya0824 this seems still missing
image

Also at:
image

File: Parsers/ASimUserManagement/Parsers/imUserManagement.yaml (hunk 23 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimUserManagement/Parsers/imUserManagement.yaml:1]: @Alekhya0824 Please re-check

[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimUserManagement/Parsers/imUserManagement.yaml:1]: image

File: Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml [hunk 25]
```diff
@@ -0,0 +1,161 @@
...
+References:
+- Title: ASIM User Management Schema
+  Link: https://aka.ms/ASimUserManagementDoc
+- Title: ASIM
+  Link: https://aka.ms/AboutASIM
+- Title: Audit User Account Management
+  Link: https://learn.microsoft.com/en-us/windows/security/threat-protection/auditing/audit-user-account-management
+Description: |
+  This ASIM parser supports normalizing Microsoft Security Event logs ingested in 'WindowsEvent' table to the ASIM User Management normalized schema.
+
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/ASimUserManagementMicrosoftWindowsEvent.yaml:19]: please remove this extra space

File: Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftWindowsEvent.yaml [hunk 26]
```diff
@@ -0,0 +1,242 @@
...
+  Version: '0.1.1'
+References:
+- Title: ASIM User Management Schema
+  Link: https://aka.ms/ASimUserManagementDoc
+- Title: ASIM
+  Link: https://aka.ms/AboutASIM
+- Title: Audit User Account Management
+  Link: https://learn.microsoft.com/en-us/windows/security/threat-protection/auditing/audit-user-account-management
+Description: |
+  This ASIM parser supports normalizing Microsoft Security Event logs ingested in 'WindowsEvent' table delivered using AMA to the ASIM UserManagement normalized schema.
```
Comment (vakohl) [inline Parsers/ASimUserManagement/Parsers/vimUserManagementMicrosoftWindowsEvent.yaml:18]: please keep same description as ASIM parser:
This ASIM parser supports normalizing Microsoft Security Event logs ingested in 'WindowsEvent' table to the ASIM User Management normalized schema.

File: Parsers/ASimUserManagement/Tests/Security_Event_ASimUserManagement_SchemaTest.csv (hunk 11 above)
Comment (Alekhya0824) [reply Parsers/ASimUserManagement/Tests/Security_Event_ASimUserManagement_SchemaTest.csv]: image

while i'm running in workspace i'm not getting this error

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**